File: Languages/Perl/math.pl
#!/usr/bin/perl

# Hello, World!
print "Hello, World!\n";

print "\nProcessing math equations...\n";

use strict;
use warnings;

sub evaluate_expression {
    my ($expr) = @_;
    
    # Remove extra whitespace
    $expr =~ s/^\s+|\s+$//g;
    
    # Use eval to solve the math equation (safe in this controlled environment)
    my $result;
    {
        no warnings 'all';
        $result = eval($expr);
    }
    
    if ($@) {
        return "Error";
    }
    
    return $result;
}

sub solve_equation {
    my ($line) = @_;
    
    my $equation = $line;
    
    # Remove "= ?" part
    $equation =~ s/=.*$//;
    
    # Trim whitespace
    $equation =~ s/^\s+|\s+$//g;
    
    my $result = evaluate_expression($equation);
    print "$equation = $result\n";
}

sub process_file {
    my ($filename) = @_;
    
    unless (-f $filename) {
        print "Could not open file: $filename\n";
        return;
    }
    
    open my $fh, '<', $filename or die "Cannot open $filename: $!";
    
    while (my $line = <$fh>) {
        chomp $line;
        
        # Skip empty lines and markdown headers
        next if $line =~ /^\s*$/ or $line =~ /^\s*#/;
        
        # Handle markdown list items
        my $start = $line;
        if ($line =~ /^\s*-\s/) {
            $start =~ s/^\s*-\s//;
        }
        
        if ($start =~ /=/) {
            solve_equation($start);
        }
    }
    
    close $fh;
}

# Process all files
process_file("../test_data/math_equations.txt");
process_file("../test_data/math_equations.md");
process_file("../test_data/math_equations.json");
process_file("../test_data/math_equations.yaml");
process_file("../test_data/math_equations");

File: Languages/PHP/math.php
<?php
// Hello, World!
echo "Hello, World!\n";

echo "\nProcessing math equations...\n";

function evaluateExpression($expr) {
    // Remove extra whitespace
    $expr = trim($expr);
    
    // Use eval to solve the math equation (safe in this controlled environment)
    try {
        return eval("return $expr;");
    } catch (ParseError $e) {
        return "Error";
    }
}

function solveEquation($line) {
    $equation = $line;
    
    // Remove "= ?" part
    $equalsPos = strpos($equation, '=');
    if ($equalsPos !== false) {
        $equation = substr($equation, 0, $equalsPos);
    }
    
    // Trim whitespace
    $equation = trim($equation);
    
    $result = evaluateExpression($equation);
    echo "$equation = $result\n";
}

function processFile($filename) {
    if (!file_exists($filename)) {
        echo "Could not open file: $filename\n";
        return;
    }
    
    $lines = file($filename, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    
    foreach ($lines as $line) {
        // Skip markdown headers
        if (strpos($line, '#') === 0) {
            continue;
        }
        
        // Handle markdown list items
        $start = $line;
        if (strpos($line, '- ') === 0) {
            $start = substr($line, 2);
        }
        
        if (strpos($start, '=') !== false) {
            solveEquation($start);
        }
    }
}

// Process all files
processFile("../test_data/math_equations.txt");
processFile("../test_data/math_equations.md");
processFile("../test_data/math_equations.json");
processFile("../test_data/math_equations.yaml");
processFile("../test_data/math_equations");
?>
